packages/financial-templates-lib/src/proxy-transaction-handler/DSProxyManager.js

```diff
@@ -0,0 +1,148 @@
+   * @param {Object} gasEstimator Module used to estimate optimal gas price with which to send txns.
+   * @param {String} account Ethereum account from which to send txns.
+   * @param {String} dsProxyFactoryAddress address of the DSProxy factory to create new DSProxies and the like.
+   */
+  constructor({ logger, web3, gasEstimator, account, dsProxyFactoryAddress, dsProxyFactoryAbi, dsProxyAbi }) {
+    assert(web3.utils.isAddress(account), "Account needs to be a valid address");
+    assert(web3.utils.isAddress(dsProxyFactoryAddress), "dsProxyFactoryAddress needs to be a valid contract address");
+    this.logger = logger;
+    this.account = account;
+    this.web3 = web3;
+    this.gasEstimator = gasEstimator;
+    this.dsProxyFactoryAddress = dsProxyFactoryAddress;
+    this.dsProxyFactory = new web3.eth.Contract(dsProxyFactoryAbi, dsProxyFactoryAddress);
+    this.dsProxyAbi = dsProxyAbi;
+    this.dsProxy = null;
+    this.dsProxyAddress = null;
+
+    // Helper functions from web3.
+    this.BN = this.web3.utils.BN;
+    this.toBN = this.web3.utils.toBN;
+    this.toWei = this.web3.utils.toWei;
+    this.fromWei = this.web3.utils.fromWei;
+    this.utf8ToHex = this.web3.utils.utf8ToHex;
+    this.isAddress = this.web3.utils.isAddress;
+
+    // Multiplier applied to Truffle's estimated gas limit for a transaction to send.
+    this.GAS_LIMIT_BUFFER = 1.25;
+  }
+
+  getDSProxyFactoryAddress() {
+    return this.dsProxyFactoryAddress;
+  }
+
+  getDSProxyAddress() {
+    return this.dsProxyAddress;
+  }
+
+  // Load in an existing DSProxy for the account EOA if one already exists OR create a new one for the user.
+  async initializeDSProxy() {
+    this.logger.debug({
+      at: "DSProxyManager",
+      message: "Initalizing...Looking for existing DSProxies or deploying a new one for the provided EOA",
+      dsProxyFactoryAddress: this.dsProxyFactoryAddress
+    });
+
+    const fromBlock = await getFromBlock(this.web3);
+    const events = await this.dsProxyFactory.getPastEvents("Created", {
+      fromBlock,
+      filter: { owner: this.account }
+    });
+
+    // The user already has a DSProxy deployed. Load it in from the events.
+    if (events.length > 0) {
+      this.dsProxyAddress = events[events.length - 1].returnValues.proxy;
+      this.dsProxy = new web3.eth.Contract(this.dsProxyAbi, this.dsProxyAddress);
+      this.logger.debug({
+        at: "DSProxyManager",
+        message: "DSProxy has been loaded in for the EOA",
+        dsProxyAddress: this.dsProxyAddress,
+        tx: events[events.length - 1].transactionHash,
+        account: this.account
+      });
+    }
+
+    // The user does not yet have a DSProxy. Create them one.
+    if (events.length == 0) {
+      this.logger.debug({
+        at: "DSProxyManager",
+        message: "No DSProxy found for EOA. Deploying new DSProxy",
+        account: this.account
+      });
+      const dsProxyCreateTx = await this.dsProxyFactory.methods.build().send({
```

Comment:
Thoughts on simulating the .send() via .estimateGas() in order to catch failed txns?

Reply:
ye, could do. let's leave that for a follow up PR though.